### PR TITLE
[GEOS-9656] JSON-LD validation will not try to validate the entire template

### PR DESCRIPTION
**Pull request by taba90**
JIRA Ticket https://osgeo-org.atlassian.net/browse/GEOS-9656

This pr fixes bugs on json-ld validation that prevent the validation process to consistely or entirely validate the json-ld template.
Moreover there are some fixes on qa failings highlighted during the attempted backport to 2.16.x of this pr https://github.com/geoserver/geoserver/pull/4248.
## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/community/json-ld/src/main/java/org/geoserver/jsonld/configuration/JsonLdConfiguration.java [hunk 1]
```diff
@@ -111,6 +111,10 @@ private NamespaceSupport declareNamespaces(FeatureType type) {
         return namespaceSupport;
     }
 
+    public GeoServerDataDirectory getDd() {
```
Comment (aaime) [inline src/community/json-ld/src/main/java/org/geoserver/jsonld/configuration/JsonLdConfiguration.java:114]: Ouch, this is really ugly. What is this, data directory? getDataDirectory then. Does it need to be public? I don't see anything else using it, is this a leftover?

Comment (taba90) [reply src/community/json-ld/src/main/java/org/geoserver/jsonld/configuration/JsonLdConfiguration.java]: fixed

**Comment by taba90** [comment]
Thanks for the review @aaime I've applied your feedback

**Comment by aaime** [comment]
Thanks, merged.

**Pull request merged**